test/TestUtilities/Microsoft.Internal.NuGet.Testing.SignedPackages/CertificateAuthority.cs

```diff
@@ -198,30 +201,53 @@ internal CertStatus GetCertStatus(CertId certId, out X509Certificate2 certificat
         {
             certificate = null;
 
-            if (certId.AlgorithmIdentifier.Algorithm.Value != Oids.Sha1)
+            Oid hashAlgorithmOid = certId.AlgorithmIdentifier.Algorithm;
+
+            if (hashAlgorithmOid.Value == Oids.Sha1)
             {
-                return CertStatus.FromUnknown();
+                // SHA-1 is obsolete.  We'll rely on serial number only for matching.
             }
-
-            if (_dnHash == null)
+            else if (IsSupportedHashAlgorithm(hashAlgorithmOid))
             {
-                using (SHA1 sha1 = SHA1.Create())
+                if (_dnHash is null || _keyHash is null)
                 {
-                    _dnHash = sha1.ComputeHash(Certificate.SubjectName.RawData);
+                    using (HashAlgorithm hashAlgorithm = CreateHashAlgorithm(hashAlgorithmOid))
+                    {
+                        _dnHash = hashAlgorithm.ComputeHash(Certificate.SubjectName.RawData);
+                        _keyHash = hashAlgorithm.ComputeHash(Certificate.GetPublicKey());
+                    }
+                }
+
+                if (!certId.IssuerNameHash.Span.SequenceEqual(_dnHash))
+                {
+                    return CertStatus.FromUnknown();
                 }
-            }
 
-            if (!certId.IssuerNameHash.Span.SequenceEqual(_dnHash))
+                if (!certId.IssuerKeyHash.Span.SequenceEqual(_keyHash))
+                {
+                    return CertStatus.FromUnknown();
+                }
+            }
+            else
             {
                 return CertStatus.FromUnknown();
             }
 
-            string serialNumber = BitConverter.ToString(certId.SerialNumber.ToByteArray()).Replace("-", "");
+            byte[] serialNumberBytes = certId.SerialNumber.ToByteArray();
+
+            // Convert to big-endian.
+            Array.Reverse(serialNumberBytes);
```

Comment:
Here's another place it would be good to make our code big endian safe, via `BitConverter.IsLittleEndian`

Reply:
From my other comment, BigInteger always uses little endian